### PR TITLE
ValueError for contradictory length of syms and dimension of tensor

### LINKED ISSUE #485: Contradictory `syms` and `arr` passed to BaseRelativityTensor not being caught
🐞 **Problem**

As mentioned in the title, the mismatch between the `syms`, which also indicates the ~order of the Tensor~ number of spacetime dimensions, and the size of the `arr` during `BaseRelativityTensor` instantiation is going unnoticed.
```python
from sympy.abc import x, y
from sympy import symbols
from einsteinpy.symbolic import BaseRelativityTensor

syms = symbols("t x y z")  
t, x, y, z= syms  
T = BaseRelativityTensor([[x, 1],[0, x+y]], syms, config="l")  
print(T.dims)  
T.tensor()  
```

🎯 **Goal**
Implement a check / error handling of inconsistency in` __init__()`

💡 **Possible solutions**

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #<put issue number here>`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

### PR DESCRIPTION
**Pull request by bibek22**
Fixes #485 

@ritzvik 
Please, see if this looks good.

### CODE REVIEW

File: src/einsteinpy/symbolic/tensor.py [hunk 1]
```diff
@@ -266,9 +266,16 @@ def __init__(
...
             Raised when argument parent_metric does not belong to MetricTensor class and isn't None.
+        ValueError
+            Raised when argument syms does not agree with shape of argument arr
 
         """
         super(BaseRelativityTensor, self).__init__(arr=arr, config=config, name=name)
+
+        if len(self.arr.shape) != 0:
+            if self.arr.shape[0] != len(syms):
+                raise ValueError(f"invalid shape of array for syms: {syms}")
```
Comment (ritzvik) [inline src/einsteinpy/symbolic/tensor.py:277]: I guess you are worried about `IndexError` if you use `and` between the 2 conditions, but it won't happen.
Replace with this, we don't want unnecessary cyclometric complexity.

```
if len(self.arr.shape) != 0 and self.arr.shape[0] != len(syms):
```

File: src/einsteinpy/symbolic/tensor.py [hunk 2]
```diff
@@ -266,9 +266,16 @@ def __init__(
             Raised when arguments syms, variables, functions have data type other than list, tuple or set.
         TypeError
             Raised when argument parent_metric does not belong to MetricTensor class and isn't None.
+        ValueError
+            Raised when argument syms does not agree with shape of argument arr
```
[the remark this answers is not in the document]
Comment (bibek22) [reply src/einsteinpy/symbolic/tensor.py]: Please feel free. I'm here to learn too. : )

File: src/einsteinpy/symbolic/tensor.py [hunk 3]
```diff
@@ -266,9 +266,15 @@ def __init__(
...
         TypeError
             Raised when argument parent_metric does not belong to MetricTensor class and isn't None.
+        ValueError
+            Raised when argument ``syms`` does not agree with shape of argument ``arr``
 
         """
         super(BaseRelativityTensor, self).__init__(arr=arr, config=config, name=name)
+
+        if len(self.arr.shape) != 0 and self.arr.shape[0] != len(syms):
+            raise ValueError(f"invalid shape of argument arr for syms: {syms}")
```
Comment (ritzvik) [inline src/einsteinpy/symbolic/tensor.py:276]: This line is incompatible with `py3.5`. I guess there would be multiple way to do this.

Please check CI logs for more details

**Comment by ritzvik** [comment]
@bibek22 everything looks good! Needs some tests here.
https://github.com/einsteinpy/einsteinpy/blob/30a68526b9cea7b21b3f81c0b89d2619c3e19440/src/einsteinpy/tests/test_symbolic/test_tensor.py

**Comment by ritzvik** [comment]
Final thing, I would ask you to squash the commits.
https://github.com/einsteinpy/einsteinpy/wiki/How-to-refactor-commits

**Comment by bibek22** [comment]
> Final thing, I would ask you to squash the commits.
> https://github.com/einsteinpy/einsteinpy/wiki/How-to-refactor-commits

This is for my own local copy of the repo, is that correct ?
I squash-rebased this particular branch with my local master before I opened the wiki link.
and then did `git pull --rebase upstream master`
Is that it ?

**Comment by ritzvik** [comment]
If you did them in local, just force push to your remote branch. The changes would be reflected here.

Regards,
Ritwik

**Comment by ritzvik** [comment]
LGTM :) Codeclimate brings up old complaints :(

**Pull request merged**